CONTRIBUTING.md

```diff
@@ -0,0 +1,41 @@
+# Contributing to Grafana's Shared Workflows Repository
+
+Welcome to the Shared Workflows repository! This repository contains reusable actions designed to streamline and automate common tasks across multiple departments and external organizations. We appreciate your interest in contributing to our shared efforts.
+
+## Types of Workflows
+
+### 1. Reusable Workflows
+Reusable workflows are located in the `.github/workflows/` directory. These workflows are primarily intended for internal use within this repository. They should be used for automating tasks specific to this repository's needs. Reusable workflows are also required if you intend to use the `on: workflow_call` event for workflow triggering. For more information on reusing workflows and using `on: workflow_call`, refer to the [GitHub documentation](https://docs.github.com/en/actions/using-workflows/reusing-workflows).
```

Comment:
> These workflows are primarily intended for internal use within this repository. They should be used for automating tasks specific to this repository's needs.

We can publish shared-workflows to be used by other repositories, for example [publish-techdocs].

[publish-techdocs]: https://github.com/grafana/shared-workflows/blob/main/.github/workflows/publish-techdocs.yaml

> Reusable workflows are also required if you intend to use the `on: workflow_call` event for workflow triggering.

I thought that's what defined something as a reusable workflow, that it can be called.

Reply:
I was trying to explain why one might add their action to the `actions/` instead of `.github/workflows/`, but I might still be a little confused on that topic. What were the reasons for us adding `publish-techdocs` as a reusable workflow, and not, for example, [push-to-gar-docker](https://github.com/grafana/shared-workflows/blob/main/actions/push-to-gar-docker/action.yaml)?

> I thought that's what defined something as a reusable workflow, that it can be called.

Can't I also call the `push-to-gar-docker` action with the `uses` stanza?

Reply:
Maybe there's some upstream documentation to delegate to so we don't have to explain this ourselves? Here's my understanding but it might be slightly off ...

Reusable workflow:

  * Called from `uses:` on a **job**
  * Specified by having `workflow_call:` in the `on:` section
  * Can itself have multiple jobs
  * Kind of like an `include` statement for creating _jobs_
  * As it's always separate jobs, and it's not that easy to share assets between different jobs in the same run, I would say these are more recommended when the work is self-contained (simple outputs are ok)

Action:

  * Called from `uses:` on a **step**
  * (for composite actions) is essentially a collection of steps, so kind of like an `include` statement for creating _steps_
  * As runs within your existing job, you can have steps before and after it, so you can share files and env. vars between your steps and the action's steps easily

> Can't I also call the push-to-gar-docker action with the uses stanza?

Not as a reusable workflow like

```yaml
job:
  foo:
    uses: ...push-to-gar-docker...

  bar:
    steps:
      - id: do-things
```

since there's no `workflow_call:` trigger (no trigger at all, since this is a composite action)

instead you use this as an action:

```yaml
job:
  foo:
    steps:
      - id: do-some-stuff
      - id: push
        uses: ...push-to-gar-docker...
      - id: do-some-other-stuff
```

Reply:
Thanks you, and I appreciate the friendly "did you try Googling that" poke :-).

I found a blog that helped explain the differences and included that in the README. WDYT?